main.py

```diff
@@ -0,0 +1,267 @@
+
+class GroundView(object):
+	def __init__(self, model):
+		self.model = model
+
+	def draw(self, surface):
+		model = self.model
+		# this takes (x,y,width,height)
+		pygame.draw.rect(surface,DIMGRAY,(model.x,model.y,model.width,model.height))
+
+class ObstacleView(object):
+	# can be used for any rectangular object
+	def __init__(self,model):
+		self.model = model
+
+	def draw(self,surface):
+		model = self.model
+		pygame.draw.rect(surface,BLACK,(model.x,model.y,model.width,model.height))
+
+"""Controller classes"""
+class Controller(object):
+	def __init__(self,models):
+		self.models = models
+		self.player = models[0] # make sure this aligns with controlled_models in main
+
+	def handle_event(self):
+		# time passed isn't actually time based... based on while loop efficiency
+		player = self.player
+		models = self.models
+		keys = pygame.key.get_pressed() # checking pressed keys
+		for model in models:
+			#movements of player
+			if keys[pygame.K_LEFT]:
+				if player.go_back():
+					model.x -= model.shiftdx
+				else:
+					model.x -= model.dx
+			if keys[pygame.K_RIGHT]:
+				if player.shift_world():
+					model.x += model.shiftdx
+				else:
+					model.x += model.dx
+		if keys[pygame.K_UP] and player.dy == 0:
+			player.dy = player.jumpdy
+
+def main():
+	pygame.init()
+	screen = pygame.display.set_mode((640,480))
+
+	# models
+	# level models:
+	ground = Ground(x=0,width=600)
+	#5 different platforms
+	platform1 = Platform(10,10)
+	platform2 = Platform(800,10)
+	platform3 = Platform(1600,10)
+	platform4 = Platform(2000,10)
+	platform5 = Platform(2400,10)
+	# player/NPC models:
+	player = Player(300,300)
+	train = PainTrain(0,300)
+	#models = [train, player, ground, platform1]
+	controlled_models = [player,train,ground,platform1,platform2,platform3,platform4,platform5]
+	level_models = [ground,platform1]
+
+	# views
+	views = []
+	views.append(PlayerView(player))
+	views.append(PainTrainView(train))
+	views.append(GroundView(ground))
+	views.append(ObstacleView(platform1))
+	views.append(ObstacleView(platform2))
+	views.append(ObstacleView(platform3))
+	views.append(ObstacleView(platform4))
+	views.append(ObstacleView(platform5))
+
+	# TODO: Add controller
+	controller = Controller(controlled_models)
+	running = True
+	counter = 0
+
+	# variable to make speed lower
+	delta_speed = .00005 # good one is .00005
+
+	while running == True:
+		# Pretty awful way to slow player down.
+		counter += 1 # adjust this if it's running to slow. Sorry.
+		if counter%5 == 0:
+			controller.handle_event()
+
+		for event in pygame.event.get():
+			if event.type == pygame.QUIT:
+				running = False
+
+		if player.train_wreck(train) or player.off_screen():
+			#Player loses if train wreck occurs or goes off screen
+			train.constdx = 0
+			player.dx = 0
+			running = False
+		#when player is falling
+		if player.hit_ground(ground):
+			player.dy = 0
+			player.y = ground.y - player.height
+		else:
+			player.dy = .3
```

Comment:
Also, this logic should probably be done in a function in model. Maybe having a super-model that contains all the player, train, and platform models, that would change the player and train's positions, and shift the world if necessary. Or you could just put it in the Player model or something.